### PR TITLE
DeleteTeam graphQL resolver

### PR DESCRIPTION
**Pull request by cbart**
Part of #47043

`DeleteTeam` deletes a given team using the corresponding database operation.

- exactly one of ID or Name must be specified
- user making the request must be site admin (at this point)
- team must exist at the time of deleting
- if Name is given for team identification, a `GetTeamByName` is issued to map name to ID.

## Test plan

Thorough unit testing.

### CODE REVIEW

File: cmd/frontend/graphqlbackend/teams.go [hunk 1]
```diff
@@ -205,8 +205,34 @@ type DeleteTeamArgs struct {
...
+		return nil, errors.New("only site admins can delete teams")
+	}
+	if args.ID == nil && args.Name == nil {
+		return nil, errors.New("team to delete is identified by either id or name, but neither was specified")
+	}
+	if args.ID != nil && args.Name != nil {
+		return nil, errors.New("team to delete is identified by either id or name, but both were specified")
+	}
+	var id int32
+	if args.ID != nil {
```
Comment (eseliger) [inline cmd/frontend/graphqlbackend/teams.go:220]: can we not use `findTeam` here as well?

File: cmd/frontend/graphqlbackend/teams_test.go [hunk 2]
```diff
@@ -467,3 +479,176 @@ func TestUpdateParentErrorBothNameAndID(t *testing.T) {
...
+			deleteTeam(name: $name) {
+				alwaysNil
+			}
+		}`,
+		ExpectedResult: `{
+			"deleteTeam": null
+		}`,
+		ExpectedErrors: []*gqlerrors.QueryError{
+			{
+				Message: "team not found: <nil>",
```
Comment (eseliger) [inline cmd/frontend/graphqlbackend/teams_test.go:616]: that error message feels weird 😀

**Review by eseliger — APPROVED** [review]

File: cmd/frontend/graphqlbackend/teams_test.go (hunk 2 above)
Comment (eseliger) [reply cmd/frontend/graphqlbackend/teams_test.go:616]: ahh got it, no prob then!

File: cmd/frontend/graphqlbackend/teams.go (hunk 1 above)
Comment (cbart) [reply cmd/frontend/graphqlbackend/teams.go:220]: That's what I done at first. Then I wanted to save the extra team lookup which you don't really need when you're deleting by ID (as `DeleteTeam` will respond with a not found). But now that you called it out - I think we can afford it.

**Pull request merged**